doc/MSTP/MSTP.md

```diff
@@ -0,0 +1,1148 @@
+            }
+
+            leaf root_address {
+                type string;
+                description
+                        "";
+            }
+
+            leaf root_cost {
+                type string;
+                description
+                        "";
+            }
+
+            leaf root_port {
+                type string;
+                description
+                        "";
+            }
+
+            leaf bridge_address {
+                type string;
+                description
+                        "";
+            }
+        }
+    }
+}
+```
+
+# Rest API Support
+Rest API is out of scope for this HLD.
+
+# Warm Boot
+Warm boot will not be supported. The IEEE 802.1s standard of MSTP does not define any potential way to support it as this might cause loops in the network. 
+
+User is expected to do a cold reboot when MSTP is running. If a user tries to perform a warm boot while MSTP is enabled, an error message will be displayed. User will first need to disable MSTP so the topology converges and reevaluates the paths.
+
+# Testing Requirements
+## Unit test cases
+### CLI test cases
+
+1. Verify CLI to enable MSTP globally. 
+1. Verify CLI to disable MSTP globally. 
+1. Verify CLI to set MSTP region name. 
+1. Verify CLI to set MSTP region revision. 
+1. Verify CLI to set MSTP port priority 
+1. Verify CLI to set MSTP hello-time 
+1. Verify CLI to set MSTP cost-value 
+1. Verify CLI to set MSTP max-hops 
+1. Verify CLI to set MSTP root guard 
+1. Verify CLI to set MSTP bpdu guard 
+1. Verify CLI to set edgeport 
+1. Verify CLI to create an instance. 
+1. Verify CLI to delete an instance 
+1. Verify CLI to clear MSTP region name. 
+1. Verify CLI to clear MSTP region revision. 
+1. Verify CLI to clear MSTP port priority 
+1. Verify CLI to clear MSTP hello-time 
+1. Verify CLI to clear MSTP cost-value 
+1. Verify CLI to clear MSTP max-hops 
+1. Verify CLI to clear MSTP root guard 
+1. Verify CLI to clear MSTP bpdu guard 
+1. Verify CLI to clear bridge priority in an instance. 
+1. Verify CLI to map VLAN to an instance 
+1. Verify CLI to delete VLAN from an instance. 
+1. Verify CLI to set interface priority on a per instance basis. 
+1. Verify CLI to set interface cost on a per instance basis. 
+1. Verify CLI to display all information related to the region. 
+1. Verify CLI to display information related to a specific instance. 
+1. Verify CLI to display information about a specific interface in a specific instance. 
+1. Verify CLI to display statistics. 
+1. Verify CLI to display statistics on a per instance basis. 
+1. Verify CLI to clear statistics. 
+1. Verify CLI to clear statistics on a per instance basis. 
+1. Verify CLI to clear statistics on a per interface per instance basis. 
+1. Verify the commands that are disabled for MSTP mode 
+
+
+### Functional Test Cases 
+
+1. Verify CONFIG DB is populated with configured MSTP parameters. 
```

Comment:
The numbering seems to be stuck with 1 (in other places too) Is it a problem with the template?

Reply:
This is the template, numbering are done automatically. if you see preview format it will show the correct format.